elixir-move.el

```diff
@@ -0,0 +1,287 @@
+            (forward-comment -1))
+          (unless (bobp)
+            (elixir-beginning-of-statement orig done limit)))
+         ;; at inline comment
+         ((looking-at "[ \t]*#")
+          (when (elixir--skip-to-semicolon-backward
+                 (save-excursion (back-to-indentation)(point)))
+            (skip-chars-forward " \t")
+            (unless (bobp)
+              (elixir-beginning-of-statement orig done limit))))
+         ;; at beginning of string
+         ((and (not done) (looking-at elixir-string-delim-re))
+          (when (< 0 (abs (skip-chars-backward " \t\r\n\f")))
+            (setq done t))
+          (back-to-indentation)
+          (elixir-beginning-of-statement orig done limit))
+         ;; after end of statement
+         ((and (not done) (eq (char-before) ?\;))
+          (skip-chars-backward ";")
+          (elixir-beginning-of-statement orig done limit))
+         ;; travel until indentation or semicolon
+         ((and (not done)
+               (elixir--skip-to-semicolon-backward
+                (save-excursion (back-to-indentation)(point))))
+          (elixir-beginning-of-statement orig done limit))
+         ;; at current indent
+         ((and (not done) (not (eq 0 (skip-chars-backward " \t\r\n\f"))))
+          (elixir-beginning-of-statement orig done limit))
+         ((and (member (char-after) (list ?\" ?\'))
+               (progn (back-to-indentation) (eq ?@ (char-after))))
+          (back-to-indentation) (setq done t)
+          (elixir-beginning-of-statement orig done limit)))
+        ;; return nil when before comment
+        (unless (and (looking-at "[ \t]*#") (looking-back "^[ \t]*"))
+          (when (< (point) orig)(setq erg (point))))
+        (when (and elixir-verbose-p (interactive-p)) (message "%s" erg))
+        erg))))
+
+(defalias 'elixir-beginning-of-statement-lc 'elixir-beginning-of-statement-bol)
+(defun elixir-beginning-of-statement-bol (&optional indent)
+  "Goto beginning of line where statement starts.
+  Returns position reached, if successful, nil otherwise.
+
+See also `elixir-up-statement': up from current definition to
+next beginning of statement above. "
+  (interactive)
+  (let* ((indent (or indent (when (eq 'elixir-end-of-statement-bol (car
+  elixir-bol-forms-last-indent))(cdr elixir-bol-forms-last-indent))))
+	 (orig (point))
+         erg)
+    (if indent
+        (while (and
+                (setq erg (elixir-beginning-of-statement))
+                (< indent (current-indentation))
+                (not (bobp))))
+      (setq erg (elixir-beginning-of-statement)))
+    ;; reset
+    (setq elixir-bol-forms-last-indent nil)
+    (beginning-of-line)
+    (and (< (point) orig) (setq erg (point)))
+    (when (interactive-p) (message "%s" erg))
+    erg))
+
+(defalias 'elixir-forward-statement 'elixir-end-of-statement)
+(defalias 'elixir-previous-statement 'elixir-end-of-statement)
+(defalias 'elixir-statement-forward 'elixir-end-of-statement)
```

Comment:
Am 27.05.2015 um 08:36 schrieb Matt DeBoard:
>
> In elixir-move.el 
> <https://github.com/elixir-lang/emacs-elixir/pull/205#discussion_r31106714>:
>
> > +    (if indent
> > +        (while (and
> > +                (setq erg (elixir-beginning-of-statement))
> > +                (< indent (current-indentation))
> > +                (not (bobp))))
> > +      (setq erg (elixir-beginning-of-statement)))
> > +    ;; reset
> > +    (setq elixir-bol-forms-last-indent nil)
> > +    (beginning-of-line)
> > +    (and (< (point) orig) (setq erg (point)))
> > +    (when (interactive-p) (message "%s" erg))
> > +    erg))
> > +
> > +(defalias 'elixir-forward-statement 'elixir-end-of-statement)
> > +(defalias 'elixir-previous-statement 'elixir-end-of-statement)
> > +(defalias 'elixir-statement-forward 'elixir-end-of-statement)
>
> What is the purpose of these aliases?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/elixir-lang/emacs-elixir/pull/205/files#r31106714>.
>

People coming from other modes might find the command via M-x apropos 
forward/previous/end RET likewise.